autoblocks/_impl/tracerv2/run.py

```diff
@@ -0,0 +1,263 @@
+import asyncio
+import contextvars
+import datetime
+import inspect
+import logging
+import threading
+from datetime import timedelta
+from typing import Any
+from typing import Coroutine
+from typing import Dict
+from typing import List
+from typing import Optional
+
+import httpx
+
+from autoblocks._impl.config.constants import INGESTION_ENDPOINT
+from autoblocks._impl.testing.models import BaseEvaluator
+from autoblocks._impl.testing.models import Evaluation
+from autoblocks._impl.tracerv2.models import SendEventResponse
+from autoblocks._impl.util import AutoblocksEnvVar
+from autoblocks._impl.util import make_replay_headers
+
+log = logging.getLogger(__name__)
+
+# Globals
+client: httpx.AsyncClient = None
+loop: asyncio.AbstractEventLoop = None
+started: bool = False
+
+
+# Context
+current_test_id_var = contextvars.ContextVar("current_test_id")
+current_test_case_var = contextvars.ContextVar("current_test_case")
+
+
+def get_or_create_ingestion_client(ingestion_key: str) -> httpx.Client:
+    global global_ingestion_client
+    if global_ingestion_client is None:
+        global_ingestion_client = httpx.Client(
+            headers={"Authorization": f"Bearer {ingestion_key}"},
+        )
```

Comment:
and then for the global client in `global_state.py` we will just do: `client = httpx.AsyncClient()` and we'll set things like headers etc when making the request